src/tlskdf.rs

```diff
@@ -14,10 +14,42 @@ use interface::*;
 use mechanism::*;
 use object::{Object, ObjectFactories};
 
-use super::bytes_to_vec;
+use super::{bytes_to_vec, cast_params};
 
 use std::fmt::Debug;
 
+macro_rules! as_ck_bbool {
+    ($key:expr, $attr:expr) => {
+        match $key.get_attr_as_bool($attr) {
+            Ok(v) => {
+                if v {
+                    CK_TRUE
+                } else {
+                    CK_FALSE
+                }
+            }
```

Comment:
ugh. Is this needed when casting bool to PKCS#11 bool?

Reply:
yeah, there is no casting in rust unless explicit casts are provide via as or into.
Remember that bool in pkcs#11 is a u8 and we need to provide a stable pointer to a u8 in CK_ATTRIBUTE, so we really need to get the right type.